docs/guides/1238-project-file-organization.md

```diff
@@ -0,0 +1,242 @@
+|-- schemas
+|   |-- compose         // <- this is a package because it includes files with a package directive
+|   |   |-- spec.cue    // package compose
+...
+|-- plan.cue            // import "example.com/pkg/schemas/compose"
+```
+
+### Summary
+
+Consider the root module as the URL to access the root of your project. Any subfolder inside this module needs to have Cue files with a package name equivalent to the dirname. The filename inside each folder is not important, the package name is:
+
+```console
+$  ls
+bar       cue.mod   foo
+
+$ cat foo/main.cue
+cat foo/main.cue 
+package foo
+
+$ cat bar/anything.cue
+package bar
+```
+
+To reference any of the code lying inside each of these packages:
+
+```cue
+import(
+    "info@foo.bar/foo" // imports CUE code inside `/foo`
+    "info@foo.bar/bar" // imports CUE code inside `/bar`
+)
+```
+
+## Initializing the root module
+
+### Project non initialized
+
+Dagger holds a way to set a module name at project initialization: `dagger project init --name <NAME>`.
+
+As said above, the name has to at least follow the structure of a domain: `domain.extension` or `domain.extension/project`. Email addresses are also accepted, and the referenced domain doesn't necessarly have to exist.
+
+In our case, let's use a fake email address, for convenience purposes:
+
+```console
+dagger project init --name "info@foo.bar"
+```
+
+### Project already initialized
+
+Manually edit the `cue.mod/module.cue` with the desired name:
+
+```console
+$ cat cue.mod/module.cue
+module: "info@foo.bar"
+```
+
+## Creating packages in subdirectories
+
+### Practice
+
+Let's put everything above into practice:
+
+- Initialize the workdir environment
+
+```console
+mkdir daggerTest && cd daggerTest
+```
+
+- Initialize the project
+
+```console
+dagger project init --name "info@foo.bar"
+```
+
+- Update `dagger` and `universe` dependencies
+
+```console
+dagger project update
+```
+
+- Create 2 subfolders: `foo` and `bar`:
+
+```console
+mkdir foo bar
+```
+
+- Inside the `bar` folder, create a CUE file with any name and whose package is `bar` (same as parent directory):
+
+```cue title="bar/anything.cue"
+package bar
+
+#Test: "world"
+```
+
+- Inside the `foo` folder, create a CUE file with any name and whose package is `foo` (same as parent directory):
+
+```cue title="foo/main.cue"
+package foo
+
+import (
+        "universe.dagger.io/alpine"
+        "universe.dagger.io/bash"
+
+        "info@foo.bar/bar"
+)
+
+#Foo: {
+    _img: alpine.#Build & {
+            packages: {
+                    bash: _
+            }
+    }
+
+    bash.#Run & {
+        env: TEST: bar.#Test
+        always: true
+        input:  _img.output
+    }
+}
+```
+
+Our `#Foo` definition is just a wrapper around `bash.#Run` that prefills the `input` field with a `bash` image (`_img` key above). Please notice that the `_img` step needs to reside inside the `#Foo` scope as it needs to exist inside the `action` to be executed.
```

Comment:
`that sets the input field to the bash image...`

`reside inside the #Foo definition so that it is available to the action context`